packages/transport/src/transports/usb.ts

```diff
@@ -0,0 +1,332 @@
+
+            const { path } = input;
+
+            const openDeviceResult = await this.transportInterface.openDevice(path, reset);
+
+            this.log('openDeviceResult', openDeviceResult);
+
+            if (!openDeviceResult.success) {
+                return reject(openDeviceResult);
+            }
+
+            const acquireDoneResult = await this.sessionsClient.acquireDone({ path });
+            this.log('acquireDoneResult', acquireDoneResult);
+
+            const expectedSessionId = acquireDoneResult.payload.session;
+            this.acquiringSession = expectedSessionId;
+
+            if (!this.listening) {
+                return resolve(this.success(expectedSessionId));
+            }
+
+            return this.listenPromise.promise
+                .then(sessionId => {
+                    resolve(this.success(sessionId));
+                    delete this.listenPromise;
+                })
+                .catch(err => {
+                    reject(this.error(err));
+                    delete this.listenPromise;
+                });
+        }, this.abortController);
+    }
+
+    call({
+        session,
+        name,
+        data,
+    }: {
+        session: string;
+        name: string;
+        data: Record<string, unknown>;
+    }) {
+        return new AbortablePromise<Success<MessageFromTrezor>, Error>(async (resolve, reject) => {
+            const getPathBySessionResponse = await this.sessionsClient.getPathBySession({
+                session,
+            });
+            if (!getPathBySessionResponse.success) {
+                // session not found means that device was disconnected
+                if (getPathBySessionResponse.error === 'session not found') {
+                    return reject(this.error({ error: 'device disconnected during action' }));
+                } else {
+                    // should never happen
+                    return reject(this.error({ error: 'unexpected error' }));
+                }
+            }
+            const { path } = getPathBySessionResponse.payload;
+
+            try {
+                await buildAndSend(
+                    this.messages,
+                    (buffer: Buffer) =>
+                        this.transportInterface.write(path, buffer).then(result => {
+                            if (!result.success) {
+                                // todo:
+                                throw new Error(result.error);
+                            }
+                        }),
+                    name,
+                    data,
+                );
+
+                const message = await receiveAndParse(this.messages, () =>
+                    this.transportInterface.read(path).then(result => {
+                        if (result.success) {
+                            return result.payload;
+                        }
+                        // todo:
+                        throw new Error(result.error);
+                    }),
```

Comment:
Maybe it would be more consistent to use only async/await syntax like:

```suggestion
const result = await this.transportInterface.read(path);
if (result.success) {
   return result.payload;
 }
throw new Error(result.error);
```